tests/src/test/java/io/jooby/FeaturedTest.java

```diff
@@ -3058,6 +3058,32 @@ public void regexpRoutingShouldIncludeDefaultValueWhenParamIsNotMatched(ServerTe
     });
   }
 
+  @ServerTest
+  public void locales(ServerTestRunner runner) {
+    runner.define(app -> {
+      app.get("/locales", Context::locales);
+      app.get("/locale", Context::locale);
+      app.setLocales(Arrays.asList(new Locale("en"),
+          new Locale("en", "GB"),
+          new Locale("de", "AT"),
+          new Locale("de", "CH"),
+          new Locale("fr"),
+          new Locale("hu")));
+    }).dontFollowRedirects().ready(client -> {
+      client.header("Accept-Language", "fr-CH, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5")
+          .get("/locales", rsp -> assertEquals("[en, en_GB, de_AT, de_CH, fr, hu]", rsp.body().string()));
```

Comment:
If `*;q=0.5` is in the header, the list returned by `Locale.filter(...)` is not sorted by priority.... have no idea why.

Reply:
It turned out that it's by design.